articles/content_filtering.md

```diff
@@ -0,0 +1,222 @@
+
+(*) These ContentFilteredTopic should not show up to user. (e.g ros2 topic list)
+
+- TimeSource
+  **use_sim_time** event on self-node_base is subscribed internally to check if **use_sim_time** parameter is changed or not. so internally this **use_sim_time** AND self node name always must be in filter_expression and expression_parameters via ContentFilteredTopic if it is supported. see [here](https://github.com/ros2/rclcpp/blob/99286978f92c30fe171313bf0785d6b6272c3257/rclcpp/src/rclcpp/time_source.cpp#L123-L125).
+  **use_sim_time** parameter event must be guaranteed by system to keep that in the filter and parameter expression when using ContentFilteredTopic.
+
+- User frontend (rclcpp/rclpy)
+  User API will be added to manage filtering configuration, so that user application can set its own filter_expression and expression_parameters. Also using AsyncParametersClient::on_parameter_event, user can take care of the parameter event with user callback only for filtered parameter events.
+  **[T.B.D] Filtering should be compatible for user even if rmw_implementation does not support ContentFilteredTopic, this can be done with using ParameterEventsFilter. But the problem here is ParameterEventsFilter only filters parameter name and event type such as NEW, DELETED, CHANGED. there is a huge gap of the flexibility that dds filtering expression and parameter expression possesses. This is what we need to discuss.**
+
+<img src="../img/content_filter/parameter_event_sequence_with_filter.png">
+
+### Interfaces
+
+#### rcl
+
+- add API `is_cft_supported` to allow frontend(rclcpp/rclpy) can know CFT is supported or not.
+  e.g) `rcl_subscription_is_cft_supported` to access `is_cft_supported`.
+- add `rcl_create_cft_subscription` to call `rmw_create_cft_subscription`.
+- add `rcl_cft_set/get_expression_parameters` to call `rmw_cft_set/get_expression_parameters`.
+
+#### rmw
+
+- add new member `is_cft_supported` into `rmw_subscription_t` which indicates if CTF is supported by rmw_implementation or not.
+  e.g) see `can_loan_masseges` member as reference.
+  support status will be set when parent subscription is created. then support status can be used to notify the upper layer if CFT is supported or not.
+
+| Implementation | `is_cft_supported` |
+| :--: | :--: |
+| rmw_connext | true |
+| rmw_fastrtps | false |
+| rmw_cyclonedds | false |
+
+- new `DDSContentFilteredTopic` object needs to be managed in implementation.
+- parent topic information should be cached so that we can create CFT based on that parent topic.
+- `rmw_create_cft_subscription` interface.
+  - input: rmw_subscription_t, rmw_node_t, parent topic information.
+  - output: rmw_subscription_t (with new data reader)
+- `rmw_cft_set/get_expression_parameters` interface.
+  - input: `DDSContentFilteredTopic`, (string sequence) expression parameters.

```

Comment:
types in `rmw` shouldn't be a DDS type.

Reply:
correct, that should be `rmw_subscriotion`, that will be fixed.